docs/src/snap/howto/cis-hardening.md

```diff
@@ -0,0 +1,290 @@
+```
+
+Test-run kube-bench against {{product}}: 
+
+```
+sudo -E kube-bench --version ck8s-dqlite-cis-1.24 --config-dir ./kube-bench-ck8s-cfg/cfg/ --config ./kube-bench-ck8s-cfg/cfg/config.yaml
+```
+
+## Harden your deployments
+
+Before running a CIS Kubernetes audit, it is essential to first harden your
+{{product}} deployment to minimize vulnerabilities and ensure 
+compliance with industry best practices, as defined by the CIS Kubernetes
+Benchmark.
+
+### Control plane nodes
+
+Run the following commands on your control plane nodes.
+
+#### Configure auditing
+
+Create an audit-policy.yaml file under `/var/snap/k8s/common/etc/` and specify 
+the level of auditing you desire based on the [upstream instructions][]. Here is 
+a minimal example of such a policy file.
+
+```
+sudo sh -c 'cat >/var/snap/k8s/common/etc/audit-policy.yaml <<EOL
+# Log all requests at the Metadata level.
+apiVersion: audit.k8s.io/v1
+kind: Policy
+rules:
+  - level: Metadata
+EOL'
+```
+
+Enable auditing at the API server level by adding the following arguments.
+
+```
+sudo sh -c 'cat >>/var/snap/k8s/common/args/kube-apiserver <<EOL
+--audit-log-path=/var/log/apiserver/audit.log
+--audit-log-maxage=30
+--audit-log-maxbackup=10
+--audit-log-maxsize=100
+--audit-policy-file=/var/snap/k8s/common/etc/audit-policy.yaml
+EOL'
+```
+
+Restart the API server:
+
+```
+sudo systemctl restart snap.k8s.kube-apiserver
+```
+
+#### Set event rate limits
+
+Create a configuration file with the [rate limits][] and place it under 
+`/var/snap/k8s/common/etc/`.
+For example:
+
+```
+sudo sh -c 'cat >/var/snap/k8s/common/etc/eventconfig.yaml <<EOL
+apiVersion: eventratelimit.admission.k8s.io/v1alpha1
+kind: Configuration
+limits:
+  - type: Server
+    qps: 5000
+    burst: 20000
+EOL'
+```
+
+Create an admissions control config file under `/var/k8s/snap/common/etc/` .
+
+```
+sudo sh -c 'cat >/var/snap/k8s/common/etc/admission-control-config-file.yaml <<EOL
+apiVersion: apiserver.config.k8s.io/v1
+kind: AdmissionConfiguration
+plugins:
+  - name: EventRateLimit
+    path: eventconfig.yaml
+EOL'
+```
+
+Make sure the EventRateLimit admission plugin is loaded in the 
+`/var/snap/k8s/common/args/kube-apiserver` .
+
+```
+--enable-admission-plugins=...,EventRateLimit,...
+```
+
+Load the admission control config file.
+
+```
+sudo sh -c 'cat >>/var/snap/k8s/common/args/kube-apiserver <<EOL
+--admission-control-config-file=/var/snap/k8s/common/etc/admission-control-config-file.yaml
+EOL'
+```
+
+Restart the API server.
+
+```
+sudo systemctl restart snap.k8s.kube-apiserver
+```
+
+#### Enable AlwaysPullImages admission control plugin
+
+Make sure the AlwaysPullImages admission plugin is loaded in the 
+`/var/snap/k8s/common/args/kube-apiserver`
+
+```
+--enable-admission-plugins=...,AlwaysPullImages,...
+```
+
+Restart the API server.
+
+```
+sudo systemctl restart snap.k8s.kube-apiserver
+```
+
+### Worker nodes
+
+Run the following commands on your worker nodes.
```

Comment:
Maybe this could help clarify that these need to be run on each worker node, and the control plane node?

Reply:
Added a line to run on both if a single node cluster